clang/test/Parser/c23-constexpr.c

```diff
@@ -0,0 +1,6 @@
+// RUN: %clang_cc1 -fsyntax-only -verify=c23 -std=c23 %s -Wpre-c2x-compat
+// RUN: %clang_cc1 -fsyntax-only -verify=c17 -std=c17 %s
+
+
```

Comment:
```suggestion

```

Reply:
We should probably add some additional parsing tests here that show we properly parse C oddities like:
```
int static constexpr i = 1; // Accepted
void func(int array[constexpr]); // Rejected (invalid syntax)
_Atomic constexpr int i = 0; // Rejected (invalid combo)
```

Reply:
Added test cases.